docs/examples/optional_questions_with_value_if_not_asked.py

```diff
@@ -0,0 +1,27 @@
+import columbo
+
+
+def user_has_dog(answers: columbo.Answers) -> bool:
+    return answers["has_dog"]
+
+
+interactions = [
+    columbo.Confirm("has_dog", "Do you have a dog?", default=True),
+    columbo.BasicQuestion(
+        "dog_name",
+        "What is the name of the dog?",
+        should_ask=user_has_dog,
+        value_if_not_asked="n/a",
+        default="Kaylee",
+    ),
+    columbo.BasicQuestion(
+        "dog_breed",
+        "What is the breed of the dog?",
+        should_ask=user_has_dog,
+        value_if_not_asked="n/a",
+        default="Basset Hound",
+    ),
+]
+
+user_answers = columbo.get_answers(interactions)
```

Comment:
Super minor, but if the default on the `Confirm` was `False` and `no_user_input=True` was set on `get_answers`, this example would return the desired output without user interaction.

Reply:
@jamescurtin : I agree with what you said - Is that the wrong behavior? The behavior you described is what I expect, but happy to entertain other views.

Reply:
Ah sorry--this thought wasn't clear. The current example is totally valid--my thought process was that switching the default value on the confirm would demonstrate the new functionality in the simplest way (_i.e._ if we wanted to be able to run the code examples from top to bottom and have it auto print out the expected result). But just sharing as a thought; no change requested

Reply:
Ah, that makes sense. Thanks for the input! I'll consider update this.